### PR TITLE
ULTIMA: Use macros for Ultima 6 detection entries

### PR DESCRIPTION
**Pull request by PushmePullyu**
- Add/fix detection entry for alt version ([#14659](https://bugs.scummvm.org/ticket/14659))
This was already added as "non-enhanced" in commit dcb4d9e91b8087056ed20a15a4b33ee2d94d97e5, but the entry for "enhanced" mode is missing.
- Add/fix detection entry for SirJohn's German fan translation v1.6
Like above: Adds missing "enhanced" entry. Normal mode was already present.
- Convert Ultima 6 detection table entries to macros to avoid repetition and make adding new versions less error-prone

### CODE REVIEW

**Comment by mduggan** [comment]
Sorry for the slow response on this one.  Looks good.

File: engines/ultima/detection_tables.h [hunk 1]
```diff
@@ -31,6 +31,37 @@ namespace Ultima {
 #define GUI_OPTIONS_MARTIAN_DREAMS GUIO0()
 #define GUI_OPTIONS_SAVAGE_EMPIRE  GUIO0()
 
+// Hack so arguments containing commas work for nested macros
+#define MACRO_ARG(...) __VA_ARGS__
```
Comment (bluegr) [inline engines/ultima/detection_tables.h:35]: Why is this needed? This is only used to wrap adFileDesc, but the comment is inconsistent with its usage

File: engines/ultima/detection_tables.h [hunk 2]
```diff
@@ -119,203 +150,58 @@ static const UltimaGameDescription GAME_DESCRIPTIONS[] = {
...
-			AD_ENTRY1s("converse.a", "5065716423ef1389e3f7b4946d815c26", 162615),
-			Common::EN_ANY,
-			Common::kPlatformDOS,
-			ADGF_NO_FLAGS,
-			GUI_OPTIONS_ULTIMA6
-		},
-		GAME_ULTIMA6,
-		GF_VGA_ENHANCED
-	},
+	ENTRY_ULTIMA6(AD_ENTRY1s("converse.a", "5065716423ef1389e3f7b4946d815c26", 162615),
```
Comment (sev-) [inline engines/ultima/detection_tables.h:153]: No, please do not do nested macros like this. Just add these three arguments as parameters as we do in other engines

**Comment by PushmePullyu** [comment]
Updated, should address both comments. Thanks for the review.

**Comment by mduggan** [comment]
Thanks!
